src/cpp/server/server_cc.cc

```diff
@@ -338,28 +338,27 @@ class Server::SyncRequestThreadManager : public ThreadManager {
   }
 
   void Shutdown() override {
-    ThreadManager::Shutdown();
     server_cq_->Shutdown();
+    ThreadManager::Shutdown();
   }
 
   void Wait() override {
-    ThreadManager::Wait();
     // Drain any pending items from the queue
     void* tag;
     bool ok;
     while (server_cq_->Next(&tag, &ok)) {
       // Do nothing
     }
+    ThreadManager::Wait();
   }
 
   void Start() {
     if (!sync_requests_.empty()) {
+      Initialize();  // ThreadManager's Initialize()
```

Comment:
Any reason for moving this from the end ?

Reply:
I thought that it didn't look nice to have requests being set up before there was an initialized thread manager ready to catch them. Additionally, just like the working of constructors, I thought that from a stylistic point of view, the base class should be prepared for action before the derived class starts its action.